### DISCUSSION TITLE
way to do breakpoint debuging or log printing?

### DISCUSSION THREAD

**Discussion by VinciShark** [soul-lang/SOUL#63] [Q&A] [answered]
I was wondering if there be a way to do breakpoint debuging or log printing?

**Comment by cesaref** [comment, marked as answer]
logging is supported in some contexts, and by the command line app (but not the website). Processors include a hidden stream `console` which if you write to, will appear in the output of the soul command. You can write any data type (including strings), and you can write multiple elements, so this is valid `console << "Test" << i << "\n";`

This will only work in processor functions, not in functions in namespaces.

As for breakpoints and general debugging, not yet, but it's on the roadmap!
